### PR TITLE
Update to CMake build

### LINKED ISSUE #258: Update build in benchcab once the updated CABLE build is released
Once the CABLE build is modified to use CMake and Spack, the build in benchcab will break. We will need to update the build in benchcab to reflect the changes.

We will have to see if benchcab should use Spack or CMake directly to build CABLE.

### PR DESCRIPTION
**Pull request by SeanBryan51**
The CABLE build system was transitioned from Makefile to CMake: see https://github.com/CABLE-LSM/CABLE/pull/216, https://github.com/CABLE-LSM/CABLE/pull/200. This change adds support for building CABLE via CMake in benchcab so that CABLE versions which branch from the HEAD of main can be tested.

Closes #258

### CODE REVIEW

**Comment by SeanBryan51** [comment]
I've decided not to add unit tests for `Model.build()` as the functionality is environment dependent and would involve a lot of unpleasant mocking. The integration test shows that the model builds successfully.

**Comment by ccarouge** [comment]
Should we remove support for the old build systems at the same time? It isn't strictly necessary for this PR but wouldn't it make things nicer for us? Bringing the old build systems to CMake might not take that much time now that we have a working example to use in main.

**Comment by SeanBryan51** [comment]
@ccarouge if you are referring to the custom build via the [`build_script`](https://benchcab.readthedocs.io/en/stable/user_guide/config_options/#build_script) key, I think I'd prefer to leave that as is for now as I would like to push a release which includes the changes in this PR. Deprecating that functionality would be major change and should require bumping the major version - we can probably save this for when spack comes around?

**Review by Whyborn — APPROVED** [review]
Looks fine to me, although I really don't know enough about the benchcab inner workings to give a detailed review.

File: src/benchcab/model.py [hunk 1]
```diff
@@ -87,7 +87,7 @@ def get_exe_path(self, mpi=False) -> Path:
         exe = internal.CABLE_MPI_EXE if mpi else internal.CABLE_EXE
         if self.install_dir:
             return internal.SRC_DIR / self.name / self.install_dir / exe
-        return internal.SRC_DIR / self.name / self.src_dir / "offline" / exe
+        return internal.SRC_DIR / self.name / "bin" / exe
```
Comment (ccarouge) [inline src/benchcab/model.py:91]: Won't this break for custom build script? Should we add something in the doc for v2 support that the executable needs to be copied to the bin/ directory?

File: src/benchcab/benchcab.py [hunk 2]
```diff
@@ -259,9 +259,7 @@ def build(self, config_path: str, mpi=False):
                 self.logger.info(
                     f"Compiling CABLE {build_mode} for realisation {repo.name}..."
                 )
```
Comment (ccarouge) [inline src/benchcab/benchcab.py:261]: Isn't this comment wrong now? It seems we compile either serially or serially and with mpi. Should we have:
```
                build_mode = "with MPI and " if mpi else ""
                self.logger.info(
                    f"Compiling CABLE {build_mode} serially for realisation {repo.name}..."
                )
```

File: src/benchcab/utils/fs.py [hunk 3]
```diff
@@ -72,3 +72,19 @@ def mkdir(new_path: Path, **kwargs):
...
+    ----------
+    var : str
+        Name of environment variable.
+    path : str
+        Path to prepend.
+    env : dict
+        Environment dictionary.
+
+    """
+    env[var] = path + (os.pathsep + env[var] if var in env else "")
```
Comment (ccarouge) [inline src/benchcab/utils/fs.py:90]: Use `env.get(var, "")` here instead of `if...else` ?

**Review by ccarouge — COMMENTED** [review]
I'm wondering if we are breaking the custom build script support but I'm not sure.

File: src/benchcab/model.py (hunk 1 above)
Comment (SeanBryan51) [reply src/benchcab/model.py:91]: The [`install_dir`](https://benchcab.readthedocs.io/en/latest/user_guide/config_options/#install_dir) key was added to address this.

Comment (SeanBryan51) [reply src/benchcab/model.py:91]: I will mention the `install_key` in the section on "Running with CABLE version 2.x", updating now.

File: src/benchcab/benchcab.py (hunk 2 above)
Comment (SeanBryan51) [reply src/benchcab/benchcab.py:261]: Done

File: src/benchcab/utils/fs.py (hunk 3 above)
Comment (SeanBryan51) [reply src/benchcab/utils/fs.py]: I've updated the implementation to:
```python
env[var] = os.pathsep.join(filter(None, [path, env.get(var)]))
```
See [dd04dd4](https://github.com/CABLE-LSM/benchcab/pull/278/commits/dd04dd4f467d5cf12eec3345a34aa0c9ce65064b)

**Review by ccarouge — APPROVED** [review]
LGTM

**Pull request merged**